pkg/kube/krt/collection.go

```diff
@@ -46,36 +44,39 @@ type dependencyState[I any] struct {
 	// Stores a map of I -> secondary dependencies (added via Fetch)
 	objectDependencies  map[Key[I]][]*dependency
 	indexedDependencies map[indexedDependency]sets.Set[Key[I]]
-	// indexedDependenciesExtractor stores a map of [collection,fetch type] => an extractor to get change keys.
-	// Note that a given collection can have multiple Fetches, but they are limited to those of the different kind.
-	// I.e. you can do a `Fetch(c1, FilterIndex()) + Fetch(c1, FilterKey())` but not `Fetch(c1, FilterIndex(idxA)) + Fetch(c1, FilterIndex(idxB))`.
-	// Multiple `FetchIndex` within a single transformation must use the same index.
-	// This only applies within a single transformation; it is fine to fetch the the same `c1` in any way from different collections.
+	// indexedDependenciesExtractor stores a map of [collection,fetch type,(optional)index id] => an extractor to get change keys.
 	indexedDependenciesExtractor map[extractorKey]objectKeyExtractor
 }
 
 type extractorKey struct {
-	uid collectionUID
-	typ indexedDependencyType
+	uid       collectionUID
+	filterUID string
```

Comment:
Is it important that filterUID be a string? It seems like we have a fair bit of conversion from collectionUID -> uint64 -> string going on but I'm not immediately seeing why.

Reply:
I thought I changed it to `collectionUID` or `uint64`, I'll update that.

Reply:
Fixed